src/utils/utils.cpp

```diff
@@ -409,6 +603,58 @@ void EchoInfo(const std::string &message) {
   }
 }
 
+void EchoStats(const std::map<std::string, std::int64_t> &stats) {
+  for (const auto &[key, value] : stats) {
+    if (value == 0) {
+      continue;
+    } else if (key == "nodes-created") {
+      std::cout << value << " nodes have been created." << std::endl;
+    } else if (key == "nodes-deleted") {
+      std::cout << value << " nodes have been deleted." << std::endl;
+    } else if (key == "relationships-created") {
+      std::cout << value << " relationships have been created." << std::endl;
+    } else if (key == "relationships-deleted") {
+      std::cout << value << " relationships have been deleted." << std::endl;
+    } else if (key == "labels-added") {
+      std::cout << value << " labels have been created." << std::endl;
+    } else if (key == "labels-removed") {
+      std::cout << value << " labels have been deleted." << std::endl;
+    } else if (key == "properties-set") {
+      std::cout << value << " properties have been updated." << std::endl;
+    }
+  }
+}
+
+void EchoNotification(const std::map<std::string, std::string> &notification) {
+  const std::string_view severity = notification.at("severity");
+  if (severity == "WARNING") {
+#ifdef _WIN32
+    HANDLE hConsole;
+    WORD original_console_attr;
+    CONSOLE_SCREEN_BUFFER_INFO csbi;
+
+    hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
+    if (!GetConsoleScreenBufferInfo(hConsole, &csbi)) {
+      std::cout << severity << ":";
+    }
+
+    original_console_attr = csbi.wAttributes;
+    FlushConsoleInputBuffer(hConsole);
+    SetConsoleTextAttribute(hConsole, FOREGROUND_RED | FOREGROUND_GREEN |
+                                          FOREGROUND_INTENSITY);
+
+    std::cout << severity << ":";
+    SetConsoleTextAttribute(hConsole, original_console_attr);
+    std::cout << " ";
```

Comment:
```suggestion
    HANDLE h_console = GetStdHandle(STD_OUTPUT_HANDLE);
    
    CONSOLE_SCREEN_BUFFER_INFO csbi;
    if (!GetConsoleScreenBufferInfo(hConsole, &csbi)) {
      const WORD original_console_attr = csbi.wAttributes;
      FlushConsoleInputBuffer(h_console);
      SetConsoleTextAttribute(h_console, FOREGROUND_RED | FOREGROUND_GREEN |
                                          FOREGROUND_INTENSITY);

      std::cout << severity << ":";
      SetConsoleTextAttribute(h_console, original_console_attr);
      std::cout << " ";
    }
```

What about consistent error handling? I think most of these functions can fail, and only `GetConsoleScreenBufferInfo` is handled in some way.

Reply:
Not sure, I've written this taking `EchoFailure` as an example. Do you think it makes sense to allow errors to propagate, if we cannot output anything is there any purpose in keeping mgconsole alive?

Reply:
I see your points and I realized why the error handling is done for `GetConsoleScreenBufferInfo` only: to make sure we don't change the console if we are sure we cannot restore it's state.

I would still change the code structure a bit in order to:
1. do not declare variables earlier than they are necessary
2. do not print `severity:` multiple times (that is a bug also in `EchoFailure`, but you can leave `EchoFailure` as is, I will create a task for it). Currently if `GetConsoleScreenBufferInfo` fails, then we still try to modify the console state and `severity:` is printed twice.

Am I missing something? If not, then I think proper behavior would be check if `GetConsoleScreenBufferInfo` succeeds, if not then just print out `serverity:` as is. If it succeeds, then do the whole console color modifications.